ocaml/typing/typedtree.ml

```diff
@@ -851,6 +858,7 @@ let as_computation_pattern (p : pattern) : computation general_pattern =
     pat_type = p.pat_type;
     pat_env = p.pat_env;
     pat_attributes = [];
+    pat_unique_barrier = fresh_unique_barrier ();
```

Comment:
Yes, that seems fine. We want to avoid unrelated patterns having the same barrier, which might lead us to be too conservative. But in this case the returned pattern should genuinely be a copy of the argument. I'll amend this.